src/middlewared/middlewared/plugins/enclosure_/map2.py

```diff
@@ -0,0 +1,59 @@
+from .slot_mappings import MAPPINGS
+
+
+def map_enclosures(enclosures):
```

Comment:
It might be appropriate to add a comment here explaining the whole `map_enclosures` mechanism, along with some explanation of the input and output of the function.